vizro-ai/pyproject.toml

```diff
@@ -18,10 +18,11 @@ dependencies = [
   "tabulate",
   "openai>=1.0.0",
   "langchain>=0.1.0, <0.3.0",  # TODO update all LLMChain class, update to pydantic v2 and remove upper bound
-  "langchain-openai",
   "langgraph>=0.1.2",
   "python-dotenv>=1.0.0",  # TODO decide env var management to see if we need this
-  "vizro>=0.1.20"
+  "vizro>=0.1.20",
+  "langchain_openai",  # Base dependency, ie minimum model working, but ideally we could get rid of this
+  "ruff"
```

Comment:
Do we want to add one more vendor in this PR? Also great if you prefer to do it in a separate PR.

```
[project.optional-dependencies]
anthropic = [
  "langchain-anthropic",
]

Reply:
See next PR - but I see the point of optional dependencies... On the docs PR I left out all but `langchain-openai`, but given that we allow for string repesentation, let me add those under optional